sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/policy/AuxiliaryAuthenticationPolicy.java

```diff
@@ -30,14 +31,17 @@
     private static final String LINKED_AUTHORIZATION_FAILED = "LinkedAuthorizationFailed";
     private static final String SCHEMA_FORMAT = "Bearer %s";
 
-    private final AzureTokenCredential[] tokenCredentials;
+    private final TokenCredential[] tokenCredentials;
+    private final AzureEnvironment environment;
 
     /**
      * Initialize an auxiliary authentication policy with the list of AzureTokenCredentials.
      *
+     * @param environment the Azure environment
      * @param credentials the AzureTokenCredentials list
      */
-    public AuxiliaryAuthenticationPolicy(AzureTokenCredential... credentials) {
+    public AuxiliaryAuthenticationPolicy(AzureEnvironment environment, TokenCredential... credentials) {
```

Comment:
what's this class used for? I think we don't need this anymore

Reply:
This is the policy used for [cross tenant authentication](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/authenticate-multi-tenant)